src/backend/distributed/utils/citus_readfuncs.c

```diff
@@ -119,6 +126,23 @@ CitusSetTag(Node *node, int tag)
 	(void) token;				/* in case not used elsewhere */ \
 	local_node->fldname = nodeRead(NULL, 0)
 
+ /* Read an integer field (anything written as ":fldname %d") */
+ #define READ_ENUM_ARRAY(fldname, count, enumtype) \
+    token = pg_strtok(&length);		/* skip :fldname */ \
+    token = pg_strtok(&length);      /* skip ( */ \
+    { \
+		int i = 0; \
+		for (i = 0; i < count; i++ ) \
+		{ \
+			token = pg_strtok(&length);		/* get field value */ \
+			local_node->fldname[i] = (enumtype) atoi(token); \
+		} \
+    } \
+	token = pg_strtok(&length);   /* skip ) */ \
+	(void) token
+
+#define READ_INT_ARRAY(fldname, count) READ_ENUM_ARRAY(fldname, count, int32)
```

Comment:
I could not find a better naming. Type cast in following line
```c
local_node->fldname[i] = (enumtype) atoi(token); \
```
requires that type is present (enum type or any signed integer).  Therefore we can't use READ_INT_ARRAY code inside READ_ENUM_ARRAY. reading int array is just a special case of reading a generic integer type with specific type in mind.